tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -2504,19 +1369,23 @@ void HWCommandQueue::enqueue_program(Program& program, bool blocking) {
     ZoneScopedN("HWCommandQueue_enqueue_program");
     std::vector<SubDeviceId> sub_device_ids = {program.determine_sub_device_ids(device)};
     TT_FATAL(sub_device_ids.size() == 1, "Programs must be executed on a single sub-device");
+    // Finalize Program: Compute relative offsets for data structures (semaphores, kernel binaries, etc) in L1
     if (not program.is_finalized()) {
         program.finalize(device);
     }
     if (program.get_program_binary_status(device->id()) == ProgramBinaryStatus::NotSent) {
-        // Write program binaries to device
+        // Write program binaries to device if it hasn't previously been cached
         program.allocate_kernel_bin_buf_on_device(device);
         if (program.get_program_transfer_info().binary_data.size()) {
             this->enqueue_write_buffer(
                 *program.get_kernels_buffer(device), program.get_program_transfer_info().binary_data.data(), false, sub_device_ids);
         }
         program.set_program_binary_status(device->id(), ProgramBinaryStatus::InFlight);
     }
-
+    // Lower the program to device: Generate dispatch commands.
+    // Values in these commands will get updated based on kernel config ring
+    // buffer state at runtime.
+    program.lower(device);
```

Comment:
Maybe have a more descriptive function name?

Reply:
Updated to `generate_dispatch_commands`.